### PR TITLE
merge up v3.8.1

### PR DESCRIPTION
**Pull request by ksunden**
- Backport changes to contribute from PR #26737
- Backport changes to contribute from PR #26737
- Backport PR #26804: Fix issue with locale comma when not using math text
- Backport PR #26807: Catch ValueError to support pytorch (and others) plotting
- Backport PR #26799: Update kiwisolver and pillow versions to be consistent with requirements
- Backport PR #26811: Add overload for slice to Spines.__getitem__
- Backport PR #26825: Fix issue with non-string labels and legend
- Backport PR #26814: Bump pypa/cibuildwheel from 2.15.0 to 2.16.0
- Backport PR #26834: Fix Issue 26821: [Bug]: ValueError: The truth value... when an ndarray is passed to the color kwarg of axes3d.scatter
- Backport PR #26843: DOC: Use ax.xaxis rather ax.get_xaxis()
- Backport PR #26843: DOC: Use ax.xaxis rather ax.get_xaxis()
- Backport PR #26849: Bump setuptools required version because of setuptools_scm v8
- Backport PR #26914: DOC: add a couple more placement examples, crosslink axes_grid [ci doc]
- Backport PR #26914: DOC: add a couple more placement examples, crosslink axes_grid [ci doc]
- Backport PR #26937: Add ArrayLike to scatter c arg type hint
- Backport PR #26927: [TYP] Remove some stubtest allowlist entries
- Backport PR #26955: Fix incorrect skip check in test_backend_ps.
- Backport PR #26952: FIX 2-tuple of colors in to_rgba_array
- Backport PR #26959: Move papersize="auto" deprecation to backend_bases.
- Backport PR #26976: Bump pypa/cibuildwheel from 2.16.0 to 2.16.1
- Backport PR #26985: Reformatted documentation under toolkits and tutorials directory
- Backport PR #26985: Reformatted documentation under toolkits and tutorials directory
- Backport PR #26850: DOC: Fix missing-reference generation on Windows
- Backport PR #26940: Add typing to pyplot.show() to avoid errors with mypy --strict.
- Backport PR #26605: ci: Install GTK4 from brew on macOS
- Backport PR #27006: DOC: Fix resizing of animation examples
- Backport PR #27006: DOC: Fix resizing of animation examples
- Backport PR #26961: DOC: made "open PR on MPL" a section in contribute guide
- Backport PR #26961: DOC: made "open PR on MPL" a section in contribute guide
- BLD: Remove development dependencies from sdists
- Backport PR #26883: [TYP] Type changes from running against Pandas
- Backport PR #27015: ValueError exception added to handle mix of {} and % string in colorbar format
- Backport PR #27017: DOC: clarify usetex versus mathtext
- Backport PR #26908: `allsegs` and `allkinds` return individual segments
- Backport PR #27043: updated api_interfaces.rst
- Backport PR #27043: updated api_interfaces.rst
- Backport PR #26995: user/project/citing updated
- Backport PR #27036: updated artist_intro.rst
- Backport PR #26995: user/project/citing updated
- Backport PR #27036: updated artist_intro.rst
- Backport PR #27054: updated interactive.rst
- Backport PR #27054: updated interactive.rst
- Backport PR #27040: Bump pypa/cibuildwheel from 2.16.1 to 2.16.2
- Backport PR #27059: ci: Clean up Python 3.12 builds
- Backport PR #26943: ci: Run mypy against typed cycler
- Backport PR #27093: [Doc]: Move Automated Tests section to workflow docs #26998
- Backport PR #27093: [Doc]: Move Automated Tests section to workflow docs #26998
- Backport PR #27096: make fonts.py, mathtext.py, text_intro.py confirm to docs guidelines
- Backport PR #27096: make fonts.py, mathtext.py, text_intro.py confirm to docs guidelines
- Backport PR #27039: Formatted docs
- Backport PR #26930: Added documentation on getting full list of registered colormaps re: issue #26244
- Backport PR #27118: Update developer release guide to follow conventions
- Backport PR #27132: changed automated tests from subsection to section in workflow
- Backport PR #27132: changed automated tests from subsection to section in workflow
- Backport PR #27153: Link xkcd color survey in named colors example
- Backport PR #27153: Link xkcd color survey in named colors example
- Backport PR #27189: Fix typo in docstring of `matplotlib.colors.from_levels_and_colors`
- Backport PR #27045: Ensure valid path mangling for ContourLabeler
- Backport PR #27179: Restore default behavior of hexbin mincnt with C provided
- Backport PR #27178: Try/except import of Axes3D
- Backport PR #27205: Improve legend picking example
- Backport PR #26970: FIX: Add PyOS_InputHook back to macos backend
- Backport PR #27088: Update `find_nearest_contour` and revert contour deprecations
- Backport PR #27217: Fix type hints for undeprecated contour APIs
- Backport PR #27221: FIX: Enable interrupts on macosx event loops
- Backport PR #27226: DOC: link out to troubleshooting guide in install
- Backport PR #27165: Fixing Matplotlib Notebook Text
- Backport PR #27237: DOC: Add command to install appropriate `requirements.txt` during dev venv setup
- Backport PR #27237: DOC: Add command to install appropriate `requirements.txt` during dev venv setup
- Release prep v3.8.1
- REL: v3.8.1
- Post v3.8.1 bump

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: doc/api/prev_api_changes/api_changes_3.8.1.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+API Changes for 3.6.1
```
Comment (tacaswell) [inline doc/api/prev_api_changes/api_changes_3.8.1.rst:1]:
```suggestion
API Changes for 3.8.1
```

Was this fixed on the -doc branch?

File: environment.yml [hunk 2]
```diff
@@ -15,10 +15,6 @@ dependencies:
   - fonttools>=4.22.0
   - importlib-resources>=3.2.0
   - kiwisolver>=1.3.1
-  - meson-python>=0.13.1
```
Comment (tacaswell) [inline environment.yml:18]: This seems wrong?

File: doc/api/prev_api_changes/api_changes_3.8.1.rst (hunk 1 above)
Comment (ksunden) [reply doc/api/prev_api_changes/api_changes_3.8.1.rst]: Yes, remerged that in here as well.

**Comment by ksunden** [comment]
Sorted out the comments, also noted that the manual backport from #27229 resulted in that paragraph repeating, but didn't cause a conflict on merge up.

**Review by tacaswell — APPROVED** [review]

File: .github/workflows/cibuildwheel.yml [hunk 3]
```diff
@@ -64,6 +64,9 @@ jobs:
         run: |
           python -m build --sdist
           python ci/export_sdist_name.py
+        env:
+          # Prevent including development runtime dependencies in metadata.
+          CIBUILDWHEEL: 1
```
Comment (QuLogic) [inline .github/workflows/cibuildwheel.yml:69]: This can be dropped, as the Meson build doesn't use the environment like setuptools did.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**